XpringKit/Xpring/XpringClient.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+
+/// Composes interactions of Xpring services.
+public class XpringClient {
+  /// A PayIDClient used to interact with the Pay ID protocol.
+  private let payIDClient: PayIDClientProtocol
+
+  /// An XRPClient used to interact with the XRP Ledger protocol.
+  private let xrpClient: XRPClientProtocol
+
+  /// Create a new XpringClient.
+  ///
+  /// - Parameters:
+  ///   - payIDClient A Pay ID Client used to interact with the Pay ID protocol.
+  ///   - xrpClient An XRP Client used to interact with the XRP Ledger protocol.
+  public init(payIDClient: PayIDClientProtocol, xrpClient: XRPClientProtocol) {
+    // TODO(keefertaylor): Validate that components are attached to the same network.
+    self.payIDClient = payIDClient
+    self.xrpClient = xrpClient
+  }
+
+  /// Send XRP to a recipient on the XRP Ledger.
+  ///
+  /// - Parameters:
+  ///    - amount: An unsigned integer representing the amount of XRP to send.
+  ///    - destinationPaymentPointer: The payment pointer which will receive the XRP.
+  ///    - sourceWallet: The wallet sending the XRP.
+  ///    - completion: A completion handler with the result of the operation.
+  /// - Throws: An error if there was a problem communicating with the XRP Ledger or the inputs were invalid.
+  // TODO(keefertaylor): Make this API synchronous to mirror functionality provided by ILP / XRP.
+  public func send(
+    _ amount: UInt64,
+    to destinationPayID: PaymentPointer,
+    from sourceWallet: Wallet,
+    completion: @escaping (Result<TransactionHash, Error>) -> Void
+  ) {
+    self.payIDClient.xrpAddress(for: destinationPayID) { [weak self] result in
```

Comment:
just a question: is this `[weak self] .. ` line in the closure basically rendering `self` to refer to the `payIDClient` ?

Reply:
Swift learning opportunity: What is `[weak self]`?

Reply:
Talked offine. This is a capture list that is preventing a retain cycle.